### PR TITLE
Support for NUnit 3

### LINKED ISSUE #1651: Akka.TestKit.NUnit doesn't work with NUnit 3.x
Tested with Akka.TestKit.NUnit 1.0.5 and NUnit 3.0.1

When running the unit tests with ReSharper test runner, the tests are discovered for a split-second but then removed and there's no indication in the UI that they were attempted to be ran. When using the NUnit 3 console runner, the tests simple do not exist.

The Akka.TestKit.NUnit package's dependency is listed as NUnit (≥ 2.6.4) which means there are no dependency issues at installation time. Maybe the Nuget dependency could be changed until a proper fix is in?

### PR DESCRIPTION
**Pull request by Silv3rcircl3**
closes #1651

### CODE REVIEW

File: src/contrib/testkits/Akka.TestKit.NUnit/packages.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="NUnit" version="2.6.4" targetFramework="net45" />
+  <package id="NUnit" version="2.6.4" targetFramework="net45" allowedVersions="(,3.0.0)"/>
```
Comment (Silv3rcircl3) [inline src/contrib/testkits/Akka.TestKit.NUnit/packages.config]: This should be < 3.0

File: src/contrib/testkits/Akka.TestKit.NUnit3.Tests/AssertionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        public AssertionsTests()
+        {
+            _assertions = new NUnitAssertions();
+        }
+
+        [Test]
+        public void Fail_should_throw()
+        {
+            Assert.Throws<AssertionException>(() => _assertions.Fail());
```
Comment (Silv3rcircl3) [inline src/contrib/testkits/Akka.TestKit.NUnit3.Tests/AssertionsTests.cs]: ```ExpectedExceptionAttribute``` was replaced with ```Assert.Throws```

File: build.cmd [hunk 3]
```diff
@@ -24,10 +24,11 @@ pushd %~dp0
 
 src\.nuget\NuGet.exe update -self
 
-src\.nuget\NuGet.exe install FAKE -ConfigFile src\.nuget\Nuget.Config -OutputDirectory src\packages -ExcludeVersion -Version 4.1.0
+src\.nuget\NuGet.exe install FAKE -ConfigFile src\.nuget\Nuget.Config -OutputDirectory src\packages -ExcludeVersion -Version 4.16.1
```
Comment (Silv3rcircl3) [inline build.cmd]: using latest FAKE version for NUnit 3 runner support

File: build.cmd [hunk 4]
```diff
@@ -24,10 +24,11 @@ pushd %~dp0
...
 
 src\.nuget\NuGet.exe install xunit.runner.console -ConfigFile src\.nuget\Nuget.Config -OutputDirectory src\packages\FAKE -ExcludeVersion -Version 2.0.0
 src\.nuget\NuGet.exe install nunit.runners -ConfigFile src\.nuget\Nuget.Config -OutputDirectory src\packages\FAKE -ExcludeVersion -Version 2.6.4
+src\.nuget\NuGet.exe install NUnit.Console -ConfigFile src\.nuget\Nuget.Config -OutputDirectory src\packages\FAKE -ExcludeVersion -Version 3.0.0
```
Comment (Silv3rcircl3) [inline build.cmd]: Nuget package for the NUnit 3 runner

**Comment by rogeralsing** [comment]
Everything builds and passes, cc @Aaronontheweb, @Horusiath do we want to pull this into the core repo? or should we start splitting things up?

**Comment by Silv3rcircl3** [comment]
By splitting things up do you mean in a separate repo inside the organization or completely outside from Akka?

**Pull request merged**